src/adapters/railgun-eth-staking.adapter.ts

```diff
@@ -0,0 +1,45 @@
+import {BindingScope, extensionFor, injectable} from '@loopback/core';
+import {BigNumber} from 'ethers';
+import {STAKING_ADAPTERS_EXTENSION_POINT} from '../keys.js';
+import {BaseStakingContractAdapter, StakingAsset} from '../staking.js';
+// Use the full path to import instead of `../types`
+import {RailgunEthStaking__factory} from '../types/factories/RailgunEthStaking__factory.js';
+
+abstract class BaseRailgunStakingContractAdapter extends BaseStakingContractAdapter {
```

Comment:
Maybe simply have `getStakedTokenBalance()` implemented in`RailgunEthereumStakingContractAdapter` class.